ProjFS.Mac/PrjFSKextTests/KauthHandlerTests.mm

```diff
@@ -32,4 +43,33 @@ - (void)testFileFlagsBitIsSet {
     XCTAssertFalse(FileFlagsBitIsSet(FileFlags_IsInVirtualizationRoot, FileFlags_Invalid));
 }
 
+- (void)testShouldIgnoreVnodeType {
+    XCTAssertTrue(ShouldIgnoreVnodeType(VNON, NULL));
+    XCTAssertTrue(ShouldIgnoreVnodeType(VBLK, NULL));
+    XCTAssertTrue(ShouldIgnoreVnodeType(VCHR, NULL));
+    XCTAssertTrue(ShouldIgnoreVnodeType(VSOCK, NULL));
+    XCTAssertTrue(ShouldIgnoreVnodeType(VFIFO, NULL));
+    XCTAssertTrue(ShouldIgnoreVnodeType(VBAD, NULL));
+    XCTAssertFalse(ShouldIgnoreVnodeType(VREG, NULL));
+    XCTAssertFalse(ShouldIgnoreVnodeType(VDIR, NULL));
+    XCTAssertFalse(ShouldIgnoreVnodeType(VLNK, NULL));
+    XCTAssertFalse(ShouldIgnoreVnodeType(VSTR, NULL));
+    XCTAssertFalse(ShouldIgnoreVnodeType(VCPLX, NULL));
+}
+
+- (void)testFileFlaggedInRoot {
+    bool fileFlaggedInRoot;
+    mockReturnBool = false;
+    mockFileFlag = FileFlags_IsInVirtualizationRoot;
+    XCTAssertFalse(TryGetFileIsFlaggedAsInRoot(reinterpret_cast<vnode_t>(static_cast<uintptr_t>(1)), NULL, &fileFlaggedInRoot));
```

Comment:
I still don't like this `reinterpret_cast` hack, although I think my previous comment about it was in one of @wilbaker's PRs, so I'll restate it here:

I'm fairly sure passing around pointers like this `reinterpret_cast<vnode_t>(static_cast<uintptr_t>(1))` is undefined behaviour, plus it's not exactly terribly readable. How about something like this at the top of the file:

```
// dummy struct definition
struct vnode
{
};

```
and then in the tests:

```
vnode test_vnode = {};
XCTAssertFalse(TryGetFileIsFlaggedAsInRoot(&test_vnode, NULL, &fileFlaggedInRoot));
```